en/24/06.md

```diff
@@ -0,0 +1,373 @@
+            self.send().direct_egld(&caller_address, &collected_fees);
+            self.collected_fees().clear();
+            }
+        }
+
+    answer: >
+      multiversx_sc::imports!();
+
+      use crate::storage;
+
+      #[multiversx_sc::module]
+      pub trait ZombieHelper: storage::Storage {
+          fn check_above_level(&self, level: u16, zombie_id: usize) {
+              let my_zombie = self.zombies(&zombie_id).get();
+              require!(my_zombie.level >= level, "Zombie is too low level");
+          }
+        
+          fn check_zombie_belongs_to_caller(&self, zombie_id: usize, caller: &ManagedAddress) {   
+            require!(
+                caller == &self.zombie_owner(&zombie_id).get(),
+                "Only the owner of the zombie can perform this operation"
+            );
+          }
+
+          #[endpoint]
+          fn change_name(&self, zombie_id: usize, name: ManagedBuffer) {
+              self.check_above_level(2u16, zombie_id);
+              let caller = self.blockchain().get_caller();
+              self.check_zombie_belongs_to_caller(zombie_id, &caller);
+              self.zombies(&zombie_id)
+                  .update(|my_zombie| my_zombie.name = name);
+          }
+
+          #[endpoint]
+          fn change_dna(&self, zombie_id: usize, dna: u64) {
+              self.check_above_level(20u16, zombie_id);
+              let caller = self.blockchain().get_caller();
+              self.check_zombie_belongs_to_caller(zombie_id, &caller);
+              self.zombies(&zombie_id)
+                  .update(|my_zombie| my_zombie.dna = dna);
+          }
+      }
+---
+
+Whoever calls our `attack` function — we want to make sure the user actually owns the zombie they're attacking with. It would be a security concern if you could attack with someone else's zombie!
+
+Can you think of how we would add a check to see if the person calling this function is the owner of the `zombie_id` they're passing in?
+
+Give it some thought, and see if you can come up with the answer on your own.
+
+Take a moment... Refer to some of our previous lessons' code for ideas...
+
+Answer below, don't continue until you've given it some thought.
+
+## The answer
+We'll go back to `zombie_helper.rs`.
+
+We've done this check multiple times now in previous lessons. In `changeName()`, `changeDna()`, and `feedAndMultiply()`, we used the following check:
+
+```
+let caller = self.blockchain().get_caller();
+require!(
+    caller == &self.zombie_owner(&zombie_id).get(),
+    "Only the owner of the zombie can perform this operation"
+);
+```
+
+This is the same logic we'll need for our `attack` function. Since we're using the same logic multiple times, let's move this into its own function to clean up our code and avoid repeating ourselves.
+
+## Put it to the test
+
+1. Create a functioncalled `check_zombie_belongs_to_caller`. It will take 2 argument, `zombie_id` (a `usuze`) and `caller` (a `&ManagedAddress`).
```

Comment:
```suggestion
1. Create a function called `check_zombie_belongs_to_caller`. It will take 2 arguments, `zombie_id` (a `usize`) and `caller` (a `&ManagedAddress`).
```